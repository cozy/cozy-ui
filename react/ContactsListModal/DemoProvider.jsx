import React from 'react'
import contacts from '../ContactsList/_mockContacts.json'
import CozyClient, { CozyProvider } from 'cozy-client'

const mockClient = new CozyClient({
  uri: 'http://cozy.tools:8080',
  token: 'faketoken'
})

mockClient.__proto__.requestQuery = ({ doctype }) => {
  if (doctype === 'io.cozy.contacts') {
    return Promise.resolve({
      fetchStatus: 'loaded',
      data: contacts
    })
  }

  if (doctype === 'io.cozy.apps') {
    return Promise.resolve({
      fetchStatus: 'loaded',
      data: [
        {
          _id: 'io.cozy.apps/contacts',
          _type: 'io.cozy.apps',
          attributes: {
            slug: 'contacts'
          },
          links: {
            related: 'http://contacts.cozy.tools:8080/'
          }
        }
      ]
    })
  }
}

mockClient.plugins = {
  realtime: {
    subscribe: () => {},
    unsubscribe: () => {},
    unsubscribeAll: () => {}
  }
}

mockClient.save = () => ({ data: contacts[0] })

const Wrapper = ({ children }) => {
  return <CozyProvider client={mockClient}>{children}</CozyProvider>
}

export default Wrapper
