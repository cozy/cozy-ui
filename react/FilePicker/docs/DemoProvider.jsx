import React from 'react'

import CozyClient, { CozyProvider } from 'cozy-client'
import { initQuery, receiveQueryResult } from 'cozy-client/dist/store'
import { normalizeDoc } from 'cozy-stack-client'

import { BreakpointsProvider } from '../../hooks/useBreakpoints'
import { I18nContext } from '../../I18n'

/**
 * @typedef {object} CreateMockClientForDocParam
 * @property {object} [queries] - Prefill queries inside the store
 */
/**
 * @typedef {object} FillQueryInsideClientQueryOptions
 * @property {IOCozyFile[]} data - Array of IOCozyFile
 * @property {{ doctype: string, selector: object }} definition - Prefill queries inside the store
 */

/**
 * @param {CozyClient} client - Instance of CozyClient
 * @param {string} queryName - Name of the query
 * @param {FillQueryInsideClientQueryOptions} queryOptions - Options of the query
 */
const fillQueryInsideClient = (client, queryName, queryOptions) => {
  const { definition, data } = queryOptions
  client.store.dispatch(initQuery(queryName, definition))
  client.store.dispatch(
    receiveQueryResult(queryName, {
      data: data ? data.map(doc => normalizeDoc(doc, definition.doctype)) : []
    })
  )
}

/**
 * Creates a client suitable for use in Documentation
 *
 * @param {CreateMockClientForDocParam} param
 * @returns {CozyClient} Instance of CozyClient
 */
const createMockClientForDoc = ({ queries = {} }) => {
  const client = new CozyClient()
  client.ensureStore()

  for (let [queryName, queryOptions] of Object.entries(queries)) {
    fillQueryInsideClient(client, queryName, queryOptions)
  }

  return client
}

const setupClient = () => {
  const client = createMockClientForDoc({
    queries: {
      'onlyfolder-io.cozy.files.root-dir': {
        data: [
          {
            _id: 'io.cozy.files.root-dir',
            id: 'io.cozy.files.root-dir',
            _type: 'io.cozy.files',
            type: 'directory',
            dir_id: 'io.cozy.files.root-dir',
            name: 'io.cozy.files.root-dir',
            updated_at: '2021-01-01T12:00:00.000000+01:00'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'io.cozy.files.root-dir'
          }
        }
      },

      'buildContentFolderQuery-io.cozy.files.root-dir': {
        data: [
          {
            _id: 'folder.01.id',
            id: 'folder.01.id',
            _type: 'io.cozy.files',
            type: 'directory',
            dir_id: 'io.cozy.files.root-dir',
            name: 'Folder one',
            updated_at: '2021-01-01T12:00:00.000000+01:00'
          },
          {
            _id: 'folder.02.id',
            id: 'folder.02.id',
            _type: 'io.cozy.files',
            type: 'directory',
            dir_id: 'io.cozy.files.root-dir',
            name: 'Folder two',
            updated_at: '2021-01-01T12:00:00.000000+01:00'
          },
          {
            _id: 'file.a.id',
            id: 'file.a.id',
            _type: 'io.cozy.files',
            type: 'file',
            dir_id: 'io.cozy.files.root-dir',
            name: 'File A.md',
            updated_at: '2021-01-01T12:00:00.000000+01:00',
            size: '111609'
          },
          {
            _id: 'file.b.id',
            id: 'file.b.id',
            _type: 'io.cozy.files',
            type: 'file',
            dir_id: 'io.cozy.files.root-dir',
            name: 'File B.png',
            updated_at: '2021-01-01T12:00:00.000000+01:00',
            size: '11160963'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'io.cozy.files.root-dir'
          }
        }
      },

      'onlyfolder-folder.01.id': {
        data: [
          {
            _id: 'folder.01.id',
            id: 'folder.01.id',
            _type: 'io.cozy.files',
            type: 'directory',
            dir_id: 'io.cozy.files.root-dir',
            name: 'Folder One',
            updated_at: '2021-01-01T12:00:00.000000+01:00'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'io.cozy.files.root-dir'
          }
        }
      },
      'buildContentFolderQuery-folder.01.id': {
        data: [
          {
            _id: 'folder.03.id',
            id: 'folder.03.id',
            _type: 'io.cozy.files',
            type: 'directory',
            dir_id: 'io.cozy.files.root-dir',
            name: 'Folder One',
            updated_at: '2021-01-01T12:00:00.000000+01:00'
          },
          {
            _id: 'file.c.id',
            id: 'file.c.id',
            _type: 'io.cozy.files',
            type: 'file',
            dir_id: 'folder.01.id',
            name: 'File C.txt',
            updated_at: '2021-01-01T12:00:00.000000+01:00',
            size: '111609'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'folder.01.id'
          }
        }
      },

      'onlyfolder-folder.02.id': {
        data: [
          {
            _id: 'folder.02.id',
            id: 'folder.02.id',
            _type: 'io.cozy.files',
            type: 'directory',
            dir_id: 'io.cozy.files.root-dir',
            name: 'Folder Two',
            updated_at: '2021-01-01T12:00:00.000000+01:00'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'io.cozy.files.root-dir'
          }
        }
      },
      'buildContentFolderQuery-folder.02.id': {
        data: [
          {
            _id: 'file.d.id',
            id: 'file.d.id',
            _type: 'io.cozy.files',
            type: 'file',
            dir_id: 'folder.02.id',
            name: 'File D.jpg',
            updated_at: '2021-01-01T12:00:00.000000+01:00',
            size: '222111'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'folder.02.id'
          }
        }
      },

      'onlyfolder-folder.03.id': {
        data: [
          {
            _id: 'folder.03.id',
            id: 'folder.03.id',
            _type: 'io.cozy.files',
            type: 'directory',
            dir_id: 'folder.01.id',
            name: 'Folder Three',
            updated_at: '2021-01-01T12:00:00.000000+01:00'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'folder.01.id'
          }
        }
      },
      'buildContentFolderQuery-folder.03.id': {
        data: [
          {
            _id: 'file.e.id',
            id: 'file.e.id',
            _type: 'io.cozy.files',
            type: 'file',
            dir_id: 'folder.03.id',
            name: 'File E.doc',
            updated_at: '2021-01-01T12:00:00.000000+01:00',
            size: '222111'
          }
        ],
        definition: {
          doctype: 'io.cozy.files',
          selector: {
            dir_id: 'folder.03.id'
          }
        }
      }
    }
  })

  return client
}

const mockClient = setupClient()

const Wrapper = ({ children }) => {
  return (
    <CozyProvider client={mockClient}>
      <BreakpointsProvider>
        <I18nContext.Provider
          value={{ t: x => x, f: () => '01 Jan. 2022', lang: 'en' }}
        >
          {children}
        </I18nContext.Provider>
      </BreakpointsProvider>
    </CozyProvider>
  )
}

export default Wrapper
