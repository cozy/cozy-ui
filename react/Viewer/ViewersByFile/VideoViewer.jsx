import React from 'react'

import withFileUrl from '../withFileUrl'

import styles from './styles.styl'

const VideoViewer = ({ file, url }) => (
  <div className={styles['viewer-videoviewer']}>
    <video src={url} controls="controls" />
    <p className={styles['viewer-filename']}>{file.name}</p>
  </div>
)

export default withFileUrl(VideoViewer)
