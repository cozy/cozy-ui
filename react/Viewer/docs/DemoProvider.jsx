import React from 'react'

import { CozyProvider } from 'cozy-client'
import { BreakpointsProvider } from '../../hooks/useBreakpoints'

import { I18nContext } from '../../I18n'

const demoTextFileResponse = {
  text: () => new Promise(resolve => resolve('Hello World !'))
}

const demoFilesByClass = {
  pdf:
    'https://raw.githubusercontent.com/mozilla/pdf.js/ba2edeae/examples/learning/helloworld.pdf',
  audio: 'https://viewerdemo.cozycloud.cc/Z.mp3',
  video: 'https://viewerdemo.cozycloud.cc/Nextcloud.mp4',
  text: 'https://viewerdemo.cozycloud.cc/notes.md'
}

const mockClient = {
  plugins: {
    realtime: {
      subscribe: () => {},
      unsubscribe: () => {},
      unsubscribeAll: () => {}
    }
  },
  on: () => {},
  collection: () => ({
    getDownloadLinkById: id =>
      new Promise(resolve => resolve(demoFilesByClass[id])),
    download: () =>
      alert(
        "This is a demo, there's no actual Cozy to download the file from ¯\\_(ツ)_/¯"
      ),
    get: () =>
      new Promise(resolve =>
        resolve({
          data: {
            links: {
              large: 'https://viewerdemo.cozycloud.cc/IMG_0062.PNG',
              preview: 'https://viewerdemo.cozycloud.cc/IMG_0062.PNG'
            }
          }
        })
      )
  }),
  getStackClient: () => ({
    uri: '',
    fetch: () => new Promise(resolve => resolve(demoTextFileResponse))
  }),
  getClient: () => mockClient,
  store: {
    getState: () => {},
    subscribe: () => {},
    unsubscribe: () => {}
  },
  getQueryFromState: () => {},
  query: () => {}
}

class Wrapper extends React.Component {
  render() {
    return (
      <CozyProvider client={mockClient}>
        <BreakpointsProvider>
          <I18nContext.Provider value={{ t: x => x, lang: 'en' }}>
            {this.props.children}
          </I18nContext.Provider>
        </BreakpointsProvider>
      </CozyProvider>
    )
  }
}

export default Wrapper
