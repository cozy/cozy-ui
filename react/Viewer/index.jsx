import ViewerContainer from './ViewerContainer'

export { default as Viewer } from './Viewer'
export { default as ViewerContainer } from './ViewerContainer'
export { toolbarPropsPropType } from './proptypes'
export {
  default as ViewerWithCustomPanelAndFooter
} from './ViewerWithCustomPanelAndFooter'

export default ViewerContainer
