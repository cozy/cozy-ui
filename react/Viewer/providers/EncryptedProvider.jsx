import React, { useContext } from 'react'

export const EncryptedContext = React.createContext()

export const useEncrypted = () => {
  const context = useContext(EncryptedContext)

  if (!context) {
    throw new Error('useEncrypted must be used within a EncryptedProvider')
  }
  return context
}

const EncryptedProvider = ({ url, children }) => {
  const contextValue = {
    url
  }
  return (
    <EncryptedContext.Provider value={contextValue}>
      {children}
    </EncryptedContext.Provider>
  )
}

export default React.memo(EncryptedProvider)
