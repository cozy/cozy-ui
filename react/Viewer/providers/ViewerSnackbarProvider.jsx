import React, { createContext, useContext, useState } from 'react'

/**
 * @typedef {('primary'|'secondary'|'success'|'error'|'warning'|'info')} enumSeverity
 */
/**
 * @typedef {object} useViewerSnackbarReturn
 * @property {boolean} isViewerSnackbarOpen
 * @property {string} viewerSnackbarMessage
 * @property {enumSeverity} viewerSnackbarSeverity
 * @property {(severity: enumSeverity, message: string) => void} showViewerSnackbar
 * @property {(message: string) => void} hideViewerSnackbar
 */

const ViewerSnackbarContext = createContext()

const ViewerSnackbarProvider = ({ children }) => {
  const [isViewerSnackbarOpen, setIsViewerSnackbarOpen] = useState(false)
  const [viewerSnackbarMessage, setViewerSnackbarMessage] = useState('')
  const [viewerSnackbarSeverity, setViewerSnackbarSeverity] = useState(
    'primary'
  )

  const showViewerSnackbar = (severity, message) => {
    setIsViewerSnackbarOpen(true)
    setViewerSnackbarMessage(message)
    setViewerSnackbarSeverity(severity)
  }
  const hideViewerSnackbar = () => {
    setIsViewerSnackbarOpen(false)
  }

  const result = {
    isViewerSnackbarOpen,
    viewerSnackbarMessage,
    viewerSnackbarSeverity,
    showViewerSnackbar,
    hideViewerSnackbar
  }

  return (
    <ViewerSnackbarContext.Provider value={result}>
      {children}
    </ViewerSnackbarContext.Provider>
  )
}

/**
 * @returns {useViewerSnackbarReturn}
 */
const useViewerSnackbar = () => {
  const viewerSnackbar = useContext(ViewerSnackbarContext)
  if (!viewerSnackbar) {
    throw new Error(
      'ViewerSnackbarContext must be used within a ViewerSnackbarProvider'
    )
  }

  return viewerSnackbar
}

export default useViewerSnackbar

export { ViewerSnackbarProvider }
