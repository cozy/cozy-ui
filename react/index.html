<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Cozy UI React components</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Inter:400,500,600">
    <link rel="stylesheet" href="main.css">
  </head>
  <body>
    <div id="rsg-root"></div>
    <script src="build/bundle.3c2f80ae.js"></script>
  </body>
</html>