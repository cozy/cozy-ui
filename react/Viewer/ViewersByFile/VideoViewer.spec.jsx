import { render, waitFor } from '@testing-library/react'
import React from 'react'

import VideoViewer from './VideoViewer'
import { BreakpointsProvider } from '../../providers/Breakpoints'
import DemoProvider from '../docs/DemoProvider'

const file = {
  _id: 'video',
  class: 'video',
  mime: 'video/mp4',
  name: 'sample.mp4'
}

const setup = () => {
  const root = render(
    <DemoProvider>
      <BreakpointsProvider>
        <VideoViewer file={file} />
      </BreakpointsProvider>
    </DemoProvider>
  )

  return { root }
}

describe('VideoViewer', () => {
  it('should render a spinner then the video viewer', async () => {
    const { root } = setup()
    const { container, queryByRole } = root

    expect(queryByRole('progressbar')).toBeTruthy()

    await waitFor(() => {
      expect(queryByRole('progressbar')).toBeFalsy()
      expect(container).toMatchSnapshot()
    })
  })
})
