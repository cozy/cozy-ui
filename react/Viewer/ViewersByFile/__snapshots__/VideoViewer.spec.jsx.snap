// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoViewer should render a spinner then the video viewer 1`] = `
<div>
  <div
    class="styles__viewer-videoviewer___NhFoe"
  >
    <video
      controls=""
      src="https://viewerdemo.cozycloud.cc/Nextcloud.mp4"
    />
    <p
      class="styles__viewer-filename___3jZCt"
    >
      sample.mp4
    </p>
  </div>
</div>
`;
